builder/scripts.py

```diff
@@ -16,15 +16,41 @@
 import os
 import sys
 
-from action import Action
+
+Action = None
+Project = None
+Import = None
+
+
+def _import_dynamic_classes():
+    # Must late import project to avoid cyclic dependency
+    global Action
+    global Project
+    global Import
+    project = __import__('project')
+    Project = getattr(project, 'Project')
+    Import = getattr(project, 'Import')
+
+    action = __import__('action')
+    Action = getattr(action, 'Action')
+
+
+def _get_all_dynamic_classes():
+    _import_dynamic_classes()
+    return set(
+        Action.__subclasses__() +
+        Project.__subclasses__() +
+        Import.__subclasses__())
 
 
 class Scripts(object):
     """ Manages loading, context, and running of per-project scripts """
-    all_actions = set()
+
+    # Must cache all classes with a reference here, or the GC will murder them
+    all_classes = set()
 
     def __init__(self):
-        Scripts.all_actions = set(Action.__subclasses__())
+        Scripts.all_classes = _get_all_dynamic_classes()
```

Comment:
Confused about static/non-static here.
`Scripts` is class with all `static` functions
`Scripts.all_classes` is class variable (static, essentiall)
But `Scripts.all_classes` gets populated in the instance constructor

When does the `Scripts.__init__()` ever get run? Is there a Scripts instance somewhere?

Reply:
Oh man you're right, it used to get instanced when we had that global builder instance. I'll fix this up.